test/fetchAndCache.test.js

```diff
@@ -0,0 +1,74 @@
+import fetchAndCache from '../src/utils/fetchAndCache';
+
+describe('fetchAndCache', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  class Response {
+    constructor(body, opts) {
+      this.body = body;
+      if (opts.headers) {
+        this.headers = {
+          get(key) {
+            return opts.headers[key];
+          },
+        };
+      }
+    }
+
+    clone() {
+      return this;
+    }
+  }
+
+  it('directly fetches if cache is not supported', async () => {
+    global.fetch = jest.fn(() => Promise.resolve('FOO'));
+    const res = await fetchAndCache('http://example.com');
+    expect(global.fetch).toHaveBeenLastCalledWith('http://example.com');
+    expect(res).toBe('FOO');
```

Comment:
You're good at writing tests! Great skill to have. 🎉 
The day I tackle adding more tests on Treeherder I will refer to this code!